pgmpy/inference/base_continuous.py

```diff
@@ -0,0 +1,268 @@
+
+    Classes inheriting this base class can be passed as an argument for
+    finding gradient log of probability distribution in inference algorithms
+
+    The class should initialize the gradient of log P(X) and log P(X)
+    (Constant terms can be ignored)
+
+    Parameters
+    ----------
+    theta : A 1d array type object or a row matrix of shape 1 X d
+            or d X 1.(Will be converted to d X 1)
+            Vector representing values of parameter theta( or X)
+
+    model : An instance of pgmpy.models.Continuous
+
+    Examples
+    --------
+    >>> # Example shows how to use the container class
+    >>> from pgmpy.inference import JointGuassianDistribution as JGD
+    >>> from pgmpy.inference import BaseGradLogPDF
+    >>> import numpy as np
+    >>> class GradLogGaussian(BaseGradLogPDF):
+    ...     def __init__(self, X, model):
+    ...         BaseGradLogPDF.__init__(self, X, model)
+    ...         self.grad_log_theta, self.log_pdf_theta = self._get_gradient_log_pdf()
+    ...     def _get_gradient_log_pdf(self):
+    ...         sub_vec = self.theta - self.model.mean_vec
+    ...         grad = - self.model.precision_matrix * sub_vec
+    ...         log_pdf = 0.5 * sub_vec.transpose * grad
+    ...         return grad, log_pdf
+    """
+
+    def __init__(self, theta, model):
+        # TODO: Take model as parameter instead of precision_matrix
+
+        if isinstance(theta, (np.matrix, np.ndarray, list, tuple, set, frozenset)):
+            theta = np.array(theta).flatten()
+            theta = np.reshape(theta, (len(theta), 1))
+        else:
+            raise TypeError("theta should be a 1d array type object")
+
+        if theta.shape[0] != model.precision_matrix.shape[0]:
+            raise ValueError("shape of theta vector should be 1 X d if shape" +
+                             " of precision matrix of model is d X d")
+        self.theta = theta
+        self.model = model
+        # The gradient of probability distribution at theta
+        self.grad_log = None
+        # The gradient log of probability distribution at theta
+        self.log_pdf = None
+
+    def get_gradient_log_pdf(self):
+        return self.grad_log, self.log_pdf
+
+
+class GradLogPDFGaussian(BaseGradLogPDF):
+    """
+    Class for finding gradient and gradient log of distribution
+    Inherits pgmpy.inference.base_continuous.BaseGradLogPDF
+    Here model must be an instance of JointGaussianDistribution
+    """
+
+    def __init__(self, theta, model):
+        BaseGradLogPDF.__init__(self, theta, model)
+        self.grad_log, self.log_pdf = self._get_gradient_log_pdf()
+
+    def _get_gradient_log_pdf(self):
+        """
+        Method that finds gradient and its log at theta
+        """
+        sub_vec = self.theta - self.model.mean_vec
+        grad = - np.dot(self.model.precision_matrix, sub_vec)
+        log_pdf = 0.5 * np.float(np.dot(sub_vec.T, grad))
+
+        return grad, log_pdf
+
+
+class BaseDiscretizeTime(object):
```

Comment:
@ankurankan Better name for this class, though in research papers and book, euler and leapfrog methods have been called methods to discretize time